File: projects/08/NameTester.java
/*###################################################
#Name: Daniela Contreras Guzman
#NameTester.java tests class Name and its Operations
#Worked with Germaine Hounakey
#Date:03/13/22
*/
import java.util.Scanner;
public class NameTester
{
public static void main(String[] args)
{
Scanner keyboard = new Scanner(System.in);
System.out.print("Testing class NameTester...");
Name aName = new Name("John", "Paul", "Jones");
// Testing get functions
assert aName.getFirst().equals("John");
assert aName.getMiddle().equals("Paul");
assert aName.getLast().equals("Jones");
assert aName.toString().equals("John Paul Jones");
assert aName.lmfi().equals("John Jones P.");
System.out.print(" 0 ");
// Testing set functions
aName.setFirst("Luz");
aName.setMiddle("Daniela");
aName.setLast("Contreras");
// get with my name 
assert aName.getFirst().equals("Luz");
assert aName.getMiddle().equals("Daniela");
assert aName.getLast().equals("Contreras");
System.out.print(" 1 ");
System.out.println("All tests passed!");

System.out.print("Enter a full name: ");
//has aFuction (first midde or last) read name frm the keyboard accordingly
aName.readName(keyboard);
System.out.println(aName.toString());
}
}
class Name {


public String getFirst() { return myFirst; }
public String getMiddle() { return myMiddle; }
public String getLast() { return myLast; }
public String toString()
{
return myFirst + ' ' + myMiddle + ' ' + myLast;
}
public String lmfi()//returns full name by segments; frst middle last
{
    return myFirst + ' ' + myLast + ' ' + myMiddle.charAt(0) + '.';
}
// Setters
    public void setFirst(String aFirst) { myFirst = aFirst; }
    public void setMiddle(String aMiddle) { myMiddle = aMiddle; }
    public void setLast(String aLast) { myLast = aLast; }
// Reader
public void readName(Scanner keyboard)//scans keyboard input from user
{
    setFirst(keyboard.next());
    setMiddle(keyboard.next());
    setLast(keyboard.next());
}
public Name(String first, String middle, String last)
{
    myFirst = first; //first name
    myMiddle = middle; //middle name
    myLast = last; //last name
}
private String myFirst, myMiddle, myLast;
};

File: projects/08/NameTester.rb
####################################################
#Namew: Daniela Contreras Guzman
#NameTester.rb tests class Name and its Operations
#Date:03/13/22
# Worked with Germaine Hounakey
####################################################
require 'test/unit/assertions' # necesssary to assert
include Test::Unit::Assertions
class Name
def initialize(first, middle, last)
@first, @middle, @last = first, middle, last
end
attr_reader :first, :middle, :last
attr_writer :first, :middle, :last
def fullName
return first + " " + middle + " " + last
end
def print
puts(fullName)
return fullName
end

# Returns "First" "Middle" "Last" as "Last First M."
def lfmi
    return last + " " + first + " " + middle[0] + "."
    end
    # Reads in each of the names into name object.
    def readName
    puts "Enter first name: "
    @first = gets.chomp
    puts "Enter middle name: "
    @middle = gets.chomp
    puts "Enter last name: "
    @last = gets.chomp
    end
    end
    def testName
    name = Name.new("John", "Paul", "Jones")

    print "Testing class nameTester... "
    assert name.first == "John", "first failed"
    assert name.middle == "Paul", "middle failed"
    assert name.last == "Jones", "last failed"
    assert name.fullName == "John Paul Jones", "fullName failed"
    print " 0 "
    name.first = "Luz"
    
    name.middle = "Daniela"
    name.last = "Contreras"
    assert name.first == "Luz", "setFirst() failed"
    assert name.middle == "Daniela", "setMiddle() failed"
    assert name.last == "Contreras", "setLast() failed"
    assert name.lfmi == "Contreras Luz D.", "lfmi() failed"
    print " 1 "
    print "All tests have successfully passed!\n"

name.readName
puts name.lfmi
end
testName